### PR TITLE
feat(component): page header alt, page header study and campaign banner alt 

### PR DESCRIPTION
**Pull request by lmounsey**
# Description

New components for page headers and campaign banners. Still doing a bit of tweaking, opening the PR so I can get a test environment so that I can test across devices/browsers a little easier.

Back in the previous docs, @jaredh-um you mentioned there is a substitute colour for the red (208, 67, 62) but I couldn't find it in the variables, so it's just baked into the component for now.  

## Component preview
[Campaign - Default](https://deploy-preview-1615--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Today%2FCampaign&selectedStory=Default)
[Campaign - Large](https://deploy-preview-1615--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Today%2FCampaign&selectedStory=Large)
[Campaign - Large with links](https://deploy-preview-1615--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Today%2FCampaign&selectedStory=Large%20%28with%20links%29)
[Campaign - Event](https://deploy-preview-1615--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Today%2FCampaign&selectedStory=Event)
[Header - Default](https://deploy-preview-1615--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Today%2FHeader&selectedStory=Default)
[Header - Default - Dark](https://deploy-preview-1615--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Today%2FHeader&selectedStory=Default%20%28Dark%29)
[Header - Study](https://deploy-preview-1615--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Today%2FHeader&selectedStory=Study)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] Make sure to do not repeat yourself (DRY)
- [x] Snapshots tested
- [x] A11y tested
- [x] CSS utilises BEM naming convention and structure
- [x] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [x] Crossbrowser testing (IE11, Safari 8+, iOS 8.4+, Android 4.4+, Firefox ESR (v52.x), iPhone (4s,6), iPad 2, Galaxy s5) 
- [x] My changes generate no new warnings
- [x] I have added tests (e2e and unit) that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Added the new component to the index.js (Vue and Lib) export

# Reviewer Checklist

- [ ] Code is not repeated (DRY)
- [ ] ES6+ code is used where possible
- [ ] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [ ] CSS utilises BEM naming convention and structure
- [ ] Snapshots tested against a copy of dev snapshots to check UI changes are intended
- [ ] A11y tested
- [ ] Crossbrowser tested in at least IE11

### CODE REVIEW

**Comment by gepifanio** [comment]
Comment: Deploy Started

Building with commit 716be537e0153e3596809ac7b78e1fff6269de94

https://app.netlify.com/sites/pattern-lib-unimelb/deploys/609a0e0d358ab60007628508

**Comment by jaredh-um** [comment]
Hey @lmounsey, just leave the red hardcoded as you have it for now. I've made this red available in a newer palette but it doesn't exist in Gen 2... but we can add a `var` for it if it ends up being repeated. Let us know when ready to review.

**Comment by lmounsey** [comment]
Hey @jaredhenderson and @gepifanio 

This one is ready for review. There is a dependency on this PR https://github.com/unimelb/pattern-lib/pull/1611 for some logic

Also, wasn't sure if you wanted me to follow existing conventions for the  Campaign banner component in regards to handling sizes - in the existing component there are `small`, `medium`, `large` sizes where as I've created an array so you can pass in anything you like

File: components/campaign/__tests__/CampaignBannerAlt.test.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import {
```
Comment (gepifanio) [inline components/campaign/__tests__/CampaignBannerAlt.test.js:1]: Could you also add an accessibility jest testing? 

Example: 
```
import { axe, toHaveNoViolations } from 'jest-axe';
  it('My Component throws no accessibility violations', (done) => {
    const html = mount(Accordion, {
      propsData: {
        name: 'title',
      },
    }).html();

    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
```

**Review by gepifanio — CHANGES_REQUESTED** [review]

**Comment by gepifanio** [comment]
Tested with 
- Site improve
- Wave
- Lighthouse 

All looks good :)

**Review by gepifanio — CHANGES_REQUESTED** [review]
Could you please also add to each story documentation how to use it as a Vue component with an example of the props, which if someone copy and paste would work straight out of the box.

Example: 

```html
  <campaign-banner-alt
    title="Register now for Open Days 2021"
    :cta="{
        text: 'Register now',
        href: '#',
      }"
    :img="{
        src: campaignHeroDesktopImage,
        alt: 'Woman typing on laptop',
        sizes: [
          {
            maxWidth: 599,
            srcset: 'url',
          },
        ],
      }" ></campaign-banner-alt>
```

* All the props that accept `object` or `array` need to have the `:` to bind the data.

**Comment by gepifanio** [comment]
- [ ] The Story with the component `PageHeaderAlt` is broken in the preview, with the images.
- [ ] Would be good to understand if we actually do need to use this progressive image component + client-only, and why not we just use standard img or div with bg image.

**Review by jaredhenderson — CHANGES_REQUESTED** [review]
- [ ] CTA button in campaign banner might need a key line around it to cater for dark backgrounds (navy on black doesn’t provide enough contrast 3:1+)
- [ ] Images aren’t loading for all Header types on the Netlify deploy as Gui has mentioned - something to do with `progressive-image` component?
- [ ] Sorry to ask, but are you able to refactor the `Header - Default Dark` component to just use a single banner image as per [these specs](https://todaydesign.atlassian.net/browse/UOM18B-101?focusedCommentId=16979) - I was asked to provide my interpretation of this banner and thought that second image under the overlay had just been added to pad out the image on the right. In any case using a single image is going to be easier to manage from a content perspective

**Comment by jaredhenderson** [comment]
- [ ] Campaign - Default with sub-text is problematic at mobile sizes - text overlapping etc. Ideally for all these campaign banners the text should not be overlapping the square device at any point, the gradient is already partially obscuring it at some viewport widths

Screen Shot 2021-05-14 at 3 51 18 pm

**Comment by lmounsey** [comment]
> * [ ]  Would be good to understand if we actually do need to use this progressive image component + client-only, and why not we just use standard img or div with bg image.

Hi @gepifanio 

The use of the progressive-img component is so that we can lazyload the image so it doesn't block the page render and speed up performance. 

For Nuxt, I had to wrap all `progressive-img` components in a nuxt only component `client-only` (which makes it only load on the client side, not server side) as Nuxt was throwing up errors in the DOM when this component was rendered server side. It should only be used on a Nuxt application, but should still work if it was included on raw HTML too.

**Comment by lmounsey** [comment]
> * [ ]  CTA button in campaign banner might need a key line around it to cater for dark backgrounds (navy on black doesn’t provide enough contrast 3:1+)
Oh yep, it was originally white on mobile so I've put that code back. Not sure if this needs to be the same on desktop

> * [x]  Images aren’t loading for all Header types on the Netlify deploy as Gui has mentioned - something to do with `progressive-image` component?
Fixed this one

> * [ ]  Sorry to ask, but are you able to refactor the `Header - Default Dark` component to just use a single banner image as per [these specs](https://todaydesign.atlassian.net/browse/UOM18B-101?focusedCommentId=16979) - I was asked to provide my interpretation of this banner and thought that second image under the overlay had just been added to pad out the image on the right. In any case using a single image is going to be easier to manage from a content perspective
I think this may be an issue on mobile because it flips vertically - this component is also coupled with the default variation on the white background so would require some refactoring.

Could we propose we identify one image to use across this component? Or, use the same image as the background?

@jaredhenderson

**Comment by lmounsey** [comment]
> * [ ]  Campaign - Default with sub-text is problematic at mobile sizes - text overlapping etc. Ideally for all these campaign banners the text should not be overlapping the square device at any point, the gradient is already partially obscuring it at some viewport widths
> 
> Screen Shot 2021-05-14 at 3 51 18 pm

We kind of threw this "subtext" in to provide a bit for flexibility on this component in the future, not really a requirement that came up - I'm wondering if we can change this banner for documentation purposes? Or I can remove this altogether - it's going to be difficult to measure how much subtext there is and make more space etc.

**Review by jaredhenderson — CHANGES_REQUESTED** [review]
> Oh yep, it was originally white on mobile so I've put that code back. Not sure if this needs to be the same on desktop

- [x] White button should be on desktop too, especially when that dark gradient is used
- [x] Simplify the Default Dark version of the header by removing the background full-width banner and any transparency from the navy overlay, so only one image (the right hand side one) is used across both header types
- [ ] The subtext banner is problematic - if no one has requested it, I'd rather it didn't exist. Users will enter screeds of text and blow out the design, so if it doesn't cater for that scenario then I'd rather it wasn't an option.

**Review by gepifanio — APPROVED** [review]
Good job 🥇

**Review by jaredhenderson — APPROVED** [review]
👍 looks good, thanks @lmounsey

**Comment by anormand** [comment]
### Campaign - Default
- SortSite - pass
- Siteimprove - pass

### Campaign - Large
- SortSite - pass
- Siteimprove - pass

### Campaign - Large with links
- SortSite - pass
- Siteimprove - pass

### Campaign - Event
- SortSite - pass
- Siteimprove - pass

### Header - Default
- SortSite - pass
- Siteimprove - pass

**Problem - image alt text.**

- The image of the student doesn't have any alt text, despite the fact that it clearly contains content
- For the same stock image image:
  - [money.visualstories.com](https://money.visualstories.com/unique-fundraising-events) went with 'conference' (at least they tried)
  - [evolution-events](https://www.evolution-events.com.mx/) went with 'Image by Evangeline Shaw'
  - [futiansourcing.com](https://futiansourcing.com/) went with 'Daily update of samples received' 

**Remedy:**
- Add alt text to the image
  
**Example:**
Change
``
To
`Business leaders at a conference`

### Header - Default - Dark
- SortSite - pass
- Siteimprove - pass

**Problem - No alt text**
- See comments on previous component

### Header - Study
- SortSite - pass
- Siteimprove - pass

**Problem - No alt text**
- The image of the student doesn't have any alt text, despite the fact that it clearly contains content

**Remedy:**
- Add alt text to the image

**Example:**
Change
``
To
`Smiling student with a backpack in front of the Old Arts Clock-tower `

**Pull request merged**